### ISSUE TITLE
Clarification on topic

### ISSUE THREAD

**Issue by beardedfool** [jneilliii/OctoPrint-TasmotaMQTT#23]
Is this expected behaviour or a possible bug?
When sonoff 4Ch Pro (6.6.0) and mqtt plugin are configured with %topic%/%prefix%/ the updating of the icon/ 

When both are configured with %prefix%/%topic%/
-the icon revolves constantly and doesn't update
- the switch doens't work as it's still trying to toggle to the last state

I presume this is why you say:
    Once installed your Tasmota devices will need to have the FullTopic configured as %topic%/%prefix%/

but then you also give a place to change this and say 
    Full Topic in plugin settings must match your relay's Full Topic pattern.

So should this full topic section in the plugin allow %prefix%/%topic%/ please?
I've too many devices working with this to change this round to suit the plugin sadly. 

Mostly thanks for making this

**Comment by jneilliii** [comment]
Yeah, the readme is outdated. You should be able to match your tasmota settings for the topic configuration. Are there any errors in the logs or browser's developer console?

**Comment by beardedfool** [comment]
Hi, thanks for coming back to me so quickly. 

Can you bear with me on this and the other issue in the non-mqtt tasmota please? I may have a deeper problem with the setup and just want to rule that out before botheing you more (fresh setup etc) - but will be after the weekend.

Just a quick, possibly connected, additional question (sorry) are you setting any retain flags at all?

I'm guessing note but don't understand python too well...

def _on_mqtt_subscription(self, topic, message, retained=**None,**

**Comment by jneilliii** [comment]
So I'm not really sure/familiar with retain in MQTT.  I was very new to the protocol when I programmed this plugin and once I got it working I kind of let it be.  Always welcome to suggestions to improve the plugin though.  The specific function you called out though is actually a callback to the MQTT plugin's helper function.  So really the retain option would be based on the topic being subscribed to I think.  Correct me if I'm wrong, but isn't the retain a part of the publishing to the topic? If that's true then the publish commands would need the retain flag set.  I could easily make that an option as we've done in the MQTT plugin.
